packages/synapse-react-client/package.json

```diff
@@ -129,10 +129,12 @@
     "@babel/runtime": "^7.19.4",
     "@babel/types": "^7.19.4",
     "@esbuild-plugins/node-modules-polyfill": "^0.2.2",
+    "@faker-js/faker": "^8.0.1",
     "@fal-works/esbuild-plugin-global-externals": "^2.1.2",
     "@googlemaps/jest-mocks": "^2.7.5",
     "@jest/reporters": "^29.5.0",
     "@juggle/resize-observer": "^3.3.1",
+    "@open-draft/deferred-promise": "^2.1.0",
```

Comment:
DefferedPromise will let us hold the response of a mock-service-worker endpoint until we're ready to send it (so we can do things like test loading states). The previous way we did this doesn't work with the `msw` upgrade.